src/components/AssetSearch/AssetSearch.tsx

```diff
@@ -29,12 +34,18 @@ export const AssetSearch = ({ onClick, filterBy }: AssetSearchProps) => {
   const searching = useMemo(() => searchString.length > 0, [searchString])
 
   useEffect(() => {
-    setFilteredAssets(
-      searching ? filterAssetsBySearchTerm(searchString, currentAssets) : currentAssets,
-    )
+    if (currentAssets) {
+      setFilteredAssets(
+        searching ? filterAssetsBySearchTerm(searchString, currentAssets) : currentAssets,
+      )
+    } else {
+      moduleLogger.error('currentAssets not defined')
+    }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [searchString])
 
+  const listAssets = searching ? filteredAssets : currentAssets
```

Comment:
We could wrap this in a `useMemo()` call to avoid getting new `listAssets` references. Probably nitpicky given the flow, as a change in `searching` will trigger re-renders anyway.